Galaxus.Functional.Tests/(Option)/OptionExtensions/UnwrapAsyncTests.cs

```diff
@@ -0,0 +1,453 @@
+﻿using System;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+namespace Galaxus.Functional.Tests.OptionExtensions
+{
+    public class UnwrapAsyncTests
+    {
+        [Test]
+        public async Task UnwrapAsync_WhenOptionIsSome_ThenCorrectValueReturned()
+        {
+            var some = Task.FromResult("hello".ToOption());
```

Comment:
I know that there's never been much of an Arrange-Act-Assert in the tests here... you think we should start doing that?

I am reading these tests and feel like I'm getting stomped by them. Funcs and Tasks always make tests more unpleasant to look at in my book, we can maybe "soften the impact" here.

What'd'ya think?